### PR TITLE
 add environment group release train

### PR DESCRIPTION
**Pull request by medo-freiheit**

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer.go [hunk 1]
```diff
@@ -827,126 +827,184 @@ func (c *DeployApplicationVersion) Transform(ctx context.Context, state *State)
...
-	var upstreamLatest = envConfig.Upstream.Latest
-	var upstreamEnvName = envConfig.Upstream.Environment
+	envGroupConfigs := make(map[string]config.EnvironmentConfig)
 
-	if !upstreamLatest && upstreamEnvName == "" {
-		return fmt.Sprintf("Environment %q does not have upstream.latest or upstream.environment configured - exiting.", targetEnvName), nil
+	for env, config := range configs {
+		if config.EnvironmentGroup != nil && *config.EnvironmentGroup == targetGroupName {
+			envGroupConfigs[env] = config
+			fmt.Println()
```
Comment (leandro-freiheit) [inline services/cd-service/pkg/repository/transformer.go]: Maybe remove?

File: services/cd-service/pkg/repository/transformer.go [hunk 2]
```diff
@@ -827,126 +827,184 @@ func (c *DeployApplicationVersion) Transform(ctx context.Context, state *State)
...
+			return fmt.Sprintf("Environment %q does not have upstream configured - exiting.", envName), nil
 		}
-	} else {
-		apps, err = state.GetEnvironmentApplications(upstreamEnvName)
-		if err != nil {
-			return "", httperrors.PublicError(ctx, fmt.Errorf("upstream environment (%q) does not have applications: %w", upstreamEnvName, err))
+		var upstreamLatest = envConfig.Upstream.Latest
+		var upstreamEnvName = envConfig.Upstream.Environment
+
+		if !upstreamLatest && upstreamEnvName == "" {
```
Comment (leandro-freiheit) [inline services/cd-service/pkg/repository/transformer.go:904]:
```suggestion
		if !upstreamLatest || upstreamEnvName == "" {
```

File: services/cd-service/pkg/repository/transformer.go (hunk 1 above)
Comment (leandro-freiheit) [reply services/cd-service/pkg/repository/transformer.go]: I mean the prinln, don't know why github tagged this whole section of code

File: services/cd-service/pkg/repository/transformer.go (hunk 2 above)
Comment (medo-freiheit) [reply services/cd-service/pkg/repository/transformer.go:904]: it’s ok if one of them is nil but not two of them, then we should throw an error

Comment (leandro-freiheit) [reply services/cd-service/pkg/repository/transformer.go:904]: You are totally right :D I was thinking of `!=` when I saw the unary operator. My bad.

**Review by leandro-freiheit — APPROVED** [review]

**Pull request merged**